onnxruntime/core/providers/rocm/rocm_execution_provider.cc

```diff
@@ -154,6 +154,25 @@ ROCMExecutionProvider::PerThreadContext::~PerThreadContext() {
   }
 }
 
+bool TunableOpIsEnabledByInfoOrEnv(const ROCMExecutionProviderInfo& info) {
+  auto env_ort_rocm_use_tunable_op = onnxruntime::GetEnvironmentVar("ORT_ROCM_USE_TUNABLE_OP");
```

Comment:
We shall remove environment variable when provider option is provided. It could lead to some issue like user forget to check/reset environment variable ...

Reply:
Ideally, I agree with you, that we should only keep session options for end users.

The problem is, in practice, when try something new with client code, it will be unexpected time wasting process, especially for deeply wrapped ort interface. Sometime they just don't have the session options properly handled. You then need to dig a stright hole from user code to the callsite of `ort.InferenceSession`. And when you want to compare it with enabled/disabled configs to show the improvement, you will need to add additional change to the parser, etc. All these changes in client code make it error prone and hard to maintain. So for the sake of ort dev productivity, lets keep it. And we shall make big warning message that these type of envs is for developer and advocate session options.

Reply:
I see. Then I suggest adding comment above this line to warn that this is for testing only, not for production use.

Reply:
I added warning level log instead of comment, so that the potential end user will be informed.